File: Go/codeWars/solutions.go
package codewars

import (
	"fmt"
	"math"
)

func GetSum(a int, b int) int {

	return (a+b)*int(math.Abs(float64(b-a))) + 1/2
}

func SimpleMultiplication(n int) int {
	if n%2 == 0 {
		return n * 8

	}
	return n * 9
}

func OddOrEven(array []int) string {
	var sum int
	for _, v := range array {
		sum += v
	}
	if sum%2 == 0 {
		return "even"
	} else {
		return "odd"
	}
}

func rowWeights(array []int) (int, int) {
	var sum1 int
	var sum2 int
	for i, v := range array {
		if i%2 == 0 {
			sum1 += v
		}
		sum2 += v
	}

	return sum1, sum2

}

func PredictAge(ages ...int) int {
	var sum int
	for _, v := range ages {
		sum += v * v
	}
	return int(math.Sqrt(float64(sum))) / 2
}

func GetMiddle(s string) string {
	middle := len(s) / 2
	if len(s)%2 == 0 {
		return s[middle-1 : middle+1]
	}
	return s[middle : middle+1]
}

func CalculateAge(birth int, year int) string {
	age := year - birth

	switch {
	case age == 0:
		return "You were born this very year!"
	case age > 0:
		return fmt.Sprintf("You are %d year%s old.", age, pluralS(age))
	default:
		return fmt.Sprintf("You will be born in %d year%s.", -age, pluralS(-age))
	}
}

func pluralS(n int) string {
	if n == 1 {
		return ""
	}
	return "s"
}

func Evaporator(content float64, evapPerDay int, threshold int) int {
	days := 0
	limit := content * (float64(threshold) / 100)
	for content > limit {
		content -= content * (float64(evapPerDay) / 100)
		days++
	}
	return days
}

func MaxAbsLengthDiff(a1 []string, a2 []string) int {
	if len(a1) == 0 || len(a2) == 0 {
		return -1
	}
	maxDiff := 0

	for _, x := range a1 {
		for _, y := range a2 {
			diff := int(math.Abs(float64(len(x) - len(y))))
			if diff > maxDiff {
				maxDiff = diff
			}
		}
	}
	return maxDiff
}

func CalculateYears(year int) (result [3]int) {
	if year < 1 {
		return
	} else if year == 1 {
		return [3]int{1, 15, 15}

	} else if year == 2 {
		return [3]int{2, 24, 24}
	} else {
		catYears := 24 + 4*(year-2)
		dogYears := 24 + 5*(year-2)
		return [3]int{year, catYears, dogYears}
	}

}

func hero(bullets, dragons int) bool {
	return bullets >= dragons*2
}

File: TypeScript/codewars/solutions.ts
function firstNonConsecutiv(arr: number[]): number | null {
  for (let i = 0; i < arr.length; i++) {
    const current = arr[i] + 1;
    if (i < arr.length - 1) {
      const next = arr[i + 1];
      if (current != next) {
        return next;
      }
    }
  }
  return null;
}

console.log(firstNonConsecutiv([1, 2, 3, 4, 6, 7, 8]));

function nameShuffler(s: string): string {
  return s.split(" ").reverse().join(" ");
}

function GetSum(a: number, b: number): number {
  return Number(((a + b) * (Math.abs(a - b) + 1)) / 2);
}

function simpleMultiplication(number: number): number {
  return number % 2 === 0 ? number * 8 : number * 9;
}

function oddOrEven(n: number[]): string {
  return n.reduce((a, b) => a + b, 0) % 2 === 0 ? "even" : "odd";
}

function rowWeigths(n: number[]): number[] {
  let a: number = 0;
  let b: number = 0;

  for (let i = 0; i < n.length; i++) {
    if (n[i] % 2 === 0) a += n[i];
    else b += n[i];
  }
  return [a, b];
}

function predictAge(...ages: number[]): number {
  return Math.floor(Math.sqrt(ages.reduce((a, b) => a + b * b, 0)) / 2);
}

console.log(predictAge(65, 60, 75, 55, 60, 63, 64, 45));

function getmiddle(s: string): string {
  const middle = s.length / 2;
  return s.length % 2 === 0
    ? s.substring(middle - 1, middle + 1)
    : s.substring(middle, middle + 1);
}

function CalculateAge(birth: number, yearTo: number): String {
  const age = yearTo - birth;
  if (age === 0) return "You were born this very year!";
  if (age > 0) return `You are ${age} year${age > 1 ? "s" : ""} old`;
  return `You will be born in ${-age} year${-age > 1 ? "s" : ""}`;
}
function evaporator(
  content: number,
  evapPerDay: number,
  threshold: number
): number {
  let days = 0;
  const limit = content * (threshold / 100);
  while (content > limit) {
    content -= content * (evapPerDay / 100);
    days++;
  }
  return days;
}

function MaxAbsLengthDiff(a1: String[], a2: String[]): number {
  if (a1.length === 0 || a2.length === 0) return -1;
  const a1Sorted = a1.sort();
  const a2Sorted = a2.sort();
  const a1SortedLength = a1Sorted.length;
  const a2SortedLength = a2Sorted.length;
  let max = 0;
  for (let i = 0; i < a1SortedLength; i++) {
    const diff = Math.abs(a1Sorted[i].length - a2Sorted[i].length);
    if (diff > max) max = diff;
  }
  return max;
}

function CalculartePetYears(humanYears: number): number[] {
  if (humanYears === 1) return [1, 15, 15];
  if (humanYears === 2) return [2, 24, 24];
  return [humanYears, 24 + (humanYears - 2) * 4, 24 + (humanYears - 2) * 5];
}
